firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -1349,8 +1349,8 @@ public void testHandlesMergeMutationWithTransformThenRemoteEvent() {
     assertContains(doc("foo/bar", 0, map("sum", 1)).setHasLocalMutations());
 
     applyRemoteEvent(addedRemoteEvent(doc("foo/bar", 1, map("sum", 1337)), asList(2), emptyList()));
-    assertChanged(doc("foo/bar", 1, map("sum", 1)).setHasLocalMutations());
-    assertContains(doc("foo/bar", 1, map("sum", 1)).setHasLocalMutations());
+    assertChanged(doc("foo/bar", 1, map("sum", 1338)).setHasLocalMutations());
```

Comment:
I am surprised that this changed. Are we not using the overlay here? The problem with this change is that it now causes flicker. If "1337" already contains "1" then we will go to 1338 and then 1337.

Reply:
It changed because the remote update changed existence state, which leads to re-applying mutation queue, without base mutations this time.

Reply:
Makes sense. We probably need to come up with a way to avoid this, but I can't think of anything out of the top of my head.